src/worker.cc

```diff
@@ -1120,6 +958,24 @@ std::string jsonify_error(const valhalla_exception_t& exception, const Api& requ
     body << (request.options().has_jsonp() ? request.options().jsonp() + "(" : "") << *json_error
          << (request.options().has_jsonp() ? ")" : "");
   }
+
+  // write a few stats about the error
+  request.mutable_info()->set_error(true);
+  auto worker = exception.code < 200 || (exception.code >= 300 && exception.code < 400)
+                    ? ".loki."
+                    : (exception.code >= 400 && exception.code <= 500 ? ".thor." : ".odin.");
+  const auto& action = Options_Action_Enum_Name(request.options().action());
+
+  auto* err_stat = request.mutable_info()->mutable_statistics()->Add();
+  err_stat->set_key(action + worker + exception.statsd_key);
+  err_stat->set_value(1);
+  err_stat->set_type(count);
+
+  auto* stat = request.mutable_info()->mutable_statistics()->Add();
+  stat->set_key(action + worker + "http_" + std::to_string(exception.http_code));
+  stat->set_value(1);
+  stat->set_type(count);
```

Comment:
here we add both the specific error metric and we send another basic metric with the http code in it. i guess this is a way to solve the question asked in: https://github.com/valhalla/valhalla/pull/3191#issuecomment-875931103

but it does double the amount of reporting we do. should we combine them into one metric?

Reply:
> but it does double the amount of reporting we do. should we combine them into one metric?

I'd lean towards combining them. Related to my comment at https://github.com/valhalla/valhalla/pull/3191/files?diff=unified&w=1#r666469902

Reply:
me too. let me think about the best way to do that and propose a change